### ISSUE TITLE
Sometimes cold-start is frozen

### ISSUE THREAD

**Issue by shakiba** [Jigsaw-Code/outline-client#1832]
**What would you like the app to enable you to do?**

Currently installing a client and sharing server configs with a new device behind internet restrictions is not easy. I have two suggestions for making it easier:

- Proxying Outline client downloads over Outline servers
- Using readable short urls for an invitation page hosted on the server (for example 142.251.32.110/YPCU8T7KGN)
- Using QR code in Outline clients (and manager) to share server configs and client download urls

**Describe alternatives you've considered**
Sometimes there is no alternative. A new device needs to already have a secure messaging app and/or a proxy/vpn tool installed.

**Additional context**

**Comment by shakiba** [comment]
Hi, thanks for picking this up!

- Yes, my suggestion was related to android sideloading.

- My suggestion is fetching it from the server itself via a temporary short url. I assumed access to the server is not restricted, otherwise the server would not be useful anyway? You would just communicate the short url verbally, so it’s helpful when you start on a device from scratch.

Your PR is a great start, thanks for your work!

**Comment by shakiba** [comment]
Why a separate IP? If it needs to be not discoverable, how about making path after the IP a random and rotating token?

**Comment by fortuna** [comment]
@shakiba just send people to the download links directly: https://www.reddit.com/r/outlinevpn/wiki/index/download_links/
We keep them on `s3.amazonaws.com` domain, so it's not blockable without blocking all of Amazon S3.

I don't understand the QR code idea. If you can send a QR code, you can send a link.

**Comment by shakiba** [comment]
> The idea behind the QR code is people can share in person without having to go over the network. You're in the same room, you pull up the QR code on your phone and your friend scans it.

Yup, QR code and short url are helpful to set up a new device that has restricted connectivity.

> In that scenario the IP is still leaked. If someone goes to the authorities saying "I got a key from x.x.x.x/<random>" then the authorities just have to block x.x.x.x

I may be missing something here, please feel free to correct me: my understanding is that when you invite someone you share the server ip address with them, so a short url to an invitation page doesn't increase the risk.

**Comment by shakiba** [comment]
Dynamic config is actually critical for updating configs, but for the first setup it’s not helpful. Eventually there are alternatives ways, but currently onboarding a new device is not very smooth.

**Comment by shakiba** [comment]
Passing the direct download link to the new device, then sharing the configs with the new client (all with very limited internet access).

**Comment by shakiba** [comment]
No. My suggestion is showing invitation when they visit <server-ip>/<readable-short-token> (and potentially expire token after a limited time).

The workaround is making a public online doc, and then using another service to make a short url or a QR code to the doc.

**Comment by fortuna** [comment]
Yes. Code here: https://github.com/fortuna/OutlineDistribution
Proof of concept: https://my.freeoutlineserver.net/

**Comment by shakiba** [comment]
Very cool, it would be great if this is integrated into Outline manager and server
